src/vma/dev/allocator.cpp

```diff
@@ -352,6 +354,13 @@ bool vma_allocator::register_memory(size_t size, ib_ctx_handler *p_ib_ctx_h,
 	return true;
 }
 
+bool vma_allocator::register_memory(void *ptr, size_t size, ib_ctx_handler *p_ib_ctx_h)
+{
+	m_data_block = ptr;
+	m_mem_alloc_type = ALLOC_TYPE_EXTERNAL;
+	return register_memory(size, p_ib_ctx_h, VMA_IBV_ACCESS_LOCAL_WRITE);
+}
+
```

Comment:
I would like to suggest changing `alloc_and_reg_mr()` to `void* alloc_and_reg_mr(size_t size, ib_ctx_handler *p_ib_ctx_h, void *ptr = NULL)` instead introducing additional `register_memory`. It looks better according current design from me.
```
	switch (m_mem_alloc_type) {
		case ALLOC_TYPE_EXTERNAL:
	m_data_block = ptr;
	m_mem_alloc_type = ALLOC_TYPE_EXTERNAL;
	register_memory(size, p_ib_ctx_h, VMA_IBV_ACCESS_LOCAL_WRITE);
```

Reply:
but then it's a bad name since it's not really allocating

Reply:
Name is not good but in this way you keep identical flow for all methods and you can say that extend alloc_and_reg_mr() function with new ability to accept user allocated memory.

Reply:
ok